blockchain/blockhashHandler.go

```diff
@@ -0,0 +1,134 @@
+package blockchain
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core/atomic"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-sdk-go/data"
+)
+
+const (
+	minPollingInterval  = time.Second
+	minBlockTtl         = time.Second
+	logPath             = "blockchain/blockhashHandler"
+	blockByHashEndpoint = "blocks/%s?fields=timestamp"
+)
+
+type argsBlockhashHandler struct {
+	pollingInterval   time.Duration
+	blockTtl          time.Duration
+	httpClientWrapper httpClientWrapper
+}
+
+type blockhashHandler struct {
+	httpClientWrapper
+	sync.RWMutex
+	log             logger.Logger
+	blockhashes     map[string]int
```

Comment:
I think it's better to use a LRU Cache instead with a fixed size.
then you wouldn't need to have a process loop for cleanup.